File: simple_functions/src/test/scala/com/jinloes/simple_functions/NumberCombinationTest.scala
package com.jinloes.simple_functions

import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FlatSpec, Matchers}

/**
  * Tests for {@link NumberCombination}
  */
@RunWith(classOf[JUnitRunner])
class NumberCombinationTest extends FlatSpec with Matchers {
  "A number combination detector" should "detect if a sum exists in an array " in {
    NumberCombination.findCombo(Array(15, 5, 3, 1), 8) should be(true)
    NumberCombination.findCombo(Array(15, 5, 3, 1), 9) should be(true)
  }

  "A number combination detector" should "return false if a number does not exist in the array" in {
    NumberCombination.findCombo(Array(15, 5, 3, 1), 10) should be(false)
    NumberCombination.findCombo(Array(15, 5, 3, 2), 1) should be(false)
    NumberCombination.findCombo(Array(15, 5, 3, 1), 17) should be(false)
  }

}

File: simple_functions/src/test/java/com/jinloes/simple_functions/NumberCombinationTest.java
package com.jinloes.simple_functions;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class NumberCombinationTest {

    @Test
    void findCombo_exists() {
        assertThat(NumberCombination.findCombo(new int[]{15, 5, 3, 1}, 8)).isTrue();
        assertThat(NumberCombination.findCombo(new int[]{15, 5, 3, 1}, 9)).isTrue();
    }

    @Test
    void findCombo_doesNotExist() {
        assertThat(NumberCombination.findCombo(new int[]{15, 5, 3, 1}, 10)).isFalse();
        assertThat(NumberCombination.findCombo(new int[]{15, 5, 3, 2}, 1)).isFalse();
        assertThat(NumberCombination.findCombo(new int[]{15, 5, 3, 1}, 17)).isFalse();
    }
}